### DISCUSSION TITLE
HDR how to wanted same picture ?

### DISCUSSION THREAD

**Discussion by ptilou** [ImageMagick/ImageMagick#6011] [Help]
### ImageMagick version

7.1.0.59

### Operating system, version and so on

Fedora 37

### Description

Hello evrybody,

I 'have more one million picture, but i've many picture same from make the HDR picture !

I wanted the script from detecting picture same, and after in bash assembly from make the Beautifull HDR ?

So i french, i don't sure my translate it's good ?
How to link the picture from easy understaint from any people ?

Great 
Philippe

**Comment by fmw42** [comment]
To find duplicates, one way is to use the magick compare tool. But that means checking every image against every other image. That is not a good way as it would be very time consuming. Alternately, you might check the hash code for the images if they are identical.

**Comment by ptilou** [comment]
Hi,

https://www.dropbox.com/scl/fo/peusccr349foy6qqgvxvq/h?dl=0&rlkey=neeo1v54e420pkkyvuhc3jcpy
This series 14 pictures, i'self selecting, it's don't same hash key include just data ...

I never use : https://imagemagick.org/script/compare.php
DO you script from detecting simular picture, or not imagemagick, but opencv or more software ?

I make this script from change the many picture to gry colorspace, it's possible auto detection by compare ? adaptation ?

```
#!/bin/bash

function f_image-fusion {
        convert  "$1"  -colorspace Gray  "$1"   "${1%}-gray.jpg"
	}
{
  find ./ -iname \*.jpg 
  echo "EOF"
		} | {
	     while true; do
	     read -r i
	     test "$i" = "EOF" && break
             f_image-fusion "$i"
        done
	}
```

Great !

Philippe

**Comment by fmw42** [comment]
if the images are the same dimensions, then try

`magick compare -metric ncc image1.suffix image2.suffix null:`

the result in parentheses will be between 0 and 1, where 1 is a perfect match and 0 is a totally bad match
